gen/gen.go

```diff
@@ -191,11 +210,33 @@ func newType(name string, fields []field) typedef {
 	}
 }
 
-func newField(name, typ, tag string) field {
+func (gc Config) newField(name, typ, tag string) field {
+	tagLine := fmt.Sprintf(`avro:"%s"`, tag)
+	for _, tc := range gc.TagConfigs {
+		if tc.Name != "avro" {
+			tagLine += fmt.Sprintf(` %s:"%s"`, tc.Name, gc.tagStyleFn(tc.Style)(tag))
+		}
```

Comment:
The nesting of the happy path is not great:
```suggestion
		if tc.Name == "avro" {
			continue
		}
		tagLine += fmt.Sprintf(` %s:"%s"`, tc.Name, gc.tagStyleFn(tc.Style)(tag))
```